pkg/cli/debug/debug.go

```diff
@@ -453,20 +456,40 @@ func (o *DebugOptions) RunDebug() error {
 			}
 			return fmt.Errorf(msg)
 			// switch to logging output
-		case err == krun.ErrPodCompleted, err == conditions.ErrContainerTerminated, !o.Attach.Stdin:
-			return logs.LogsOptions{
-				Object: pod,
-				Options: &corev1.PodLogOptions{
-					Container: o.Attach.ContainerName,
-					Follow:    true,
-				},
-				RESTClientGetter: o.RESTClientGetter,
-				ConsumeRequestFn: logs.DefaultConsumeRequest,
-				IOStreams:        o.IOStreams,
-				LogsForObject:    o.LogsForObject,
-			}.RunLogs()
+		case err == krun.ErrPodCompleted, err == conditions.ErrContainerTerminated:
+			o.logsOptions(pod)
+		case err == conditions.ErrNonZeroExitCode:
+			r, w := io.Pipe()
+			streams := genericclioptions.IOStreams{
+				In:     r,
+				Out:    w,
+				ErrOut: ioutil.Discard,
+			}
+			go func() {
+				fmt.Fprintf(w, "%v", logs.LogsOptions{
```

Comment:
I see what you're trying to do here, but frankly said I don't like it. It has to be cleaner.

Reply:
yea, i'm capturing the stdout and creating the error when appropriate (if container exits non-zero)

Reply:
I don't like it much either, but the LogsOptions will return nil error as long as it completes, so the cmd will never return non-zero unless LogsOptions fails. I think the goal is for oc debug to return non-zero if the container returns non-zero

Reply:
Let's split this PR, this bit that's questionable - open a separate PR for it, the rest ([BZ#1771552](https://bugzilla.redhat.com/show_bug.cgi?id=1771552) and [BZ#1771553](https://bugzilla.redhat.com/show_bug.cgi?id=1771553)) leave here.

Reply:
see my comment above, please take another look : )